### ISSUE TITLE
Memory issue

### ISSUE THREAD

**Issue by phpepe** [VadimDez/ng2-pdf-viewer#392]
##### Bug Report or Feature Request (mark with an `x`)
```
- [ ] Regression (a behavior that used to work and stopped working in a new release)
- [x] Bug report -> please search issues before submitting
- [ ] Feature request
- [ ] Documentation issue or request
```
Using under ionic 3 application, on ios 12, got the following memory issue after increasing the zoom:
ng2-pdf-viewer version: 5.0.1

```
      total canvas memory use exceeds the maximum limit (224 mb)
```

The source file weight its around 700kb which I think is accurate. 
Any help its appreciated

**Comment by phpepe** [comment]
Thanks i will try it !

**Comment by phpepe** [comment]
I noticed your PR has been accepted, which is a great news. 
Also after some text, it seems like works better.
It allows me to increment the zoom factor until certain number (1.5x, 2x) but still reaches the memory limit after that. 
To solve  completely that limitation for large PDF files I guess that some type of lazy loading over canvas elements needs to be done, and not one canvas per page as is now. 
Do you know how difficult is that implementation  ?
I may be able to help on that after getting familiar with this codebase

**Comment by AntonioDell** [comment]
I think this would require a new implementation of the pdfjs/pdf-viewer which is currently used to display the pdf file. Since all zooming is just passed to that viewer, I don't think, that it's an issue, that can be resolved here.
You could try to limit the zooming to the factor 2.0 (in my opinion, more zooming shouldn't be needed in any case).
To avoid this issue you could check, if your pdf file is optimized for web viewing ("linearization" is the word oyu should search for in thix context).

**Comment by KalaivananE** [comment]
i am using ng2 pdf viewer for render pdf in electron application.When i put a pdf file with large size of width and height it renders very slow.
can anyone suggest the solution for that.
take this example file.
architectural_-_annotation_scaling_and_multileaders.SECTIONS AND DETAILSs2 (2).png.pdf
or
https://github.com/mozilla/pdf.js/issues/9581

**Comment by erata** [comment]
hi, i use pdf viewer 2.0.2 wit angular 4. when try to zoom in pdf in iphone 7 plus it gives memory error such as above.  how can i solve the problem?

**Comment by VadimDez** [comment]
Fixed in `6.0.0` See https://github.com/VadimDez/ng2-pdf-viewer/blob/master/CHANGELOG.md#600
